File: app/src/main/java/com/example/petridelivery/fragments/abs/BaseFragment.java
package com.example.petridelivery.fragments.abs;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.petridelivery.abs.BaseMainActivity;
import com.example.petridelivery.app.PetriDeliveryApp;
import com.example.petridelivery.wrappers.WrapperComponent;

public abstract class BaseFragment extends Fragment {

    protected WrapperComponent wc;
    protected PetriDeliveryApp app;
    protected BaseMainActivity activity;

    public BaseFragment(int contentLayoutId) {
        super(contentLayoutId);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setup();
    }

    protected void setup(){
        activity = (BaseMainActivity) getActivity();

        wc = activity.getWc();
        app = activity.getApp();
    }

}

File: app/src/main/java/com/example/petridelivery/fragments/abs/BaseFragment.kt
package com.example.petridelivery.fragments.abs

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import com.example.petridelivery.abs.BaseMainActivity
import com.example.petridelivery.app.PetriDeliveryApp
import com.example.petridelivery.wrappers.WrapperComponent

abstract class BaseFragment(contentLayoutId: Int) : Fragment(contentLayoutId) {
    protected lateinit var wc: WrapperComponent
    protected lateinit var app: PetriDeliveryApp
    protected lateinit var activity: BaseMainActivity

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setup()
    }

    protected fun setup() {
        activity = getActivity() as BaseMainActivity
        wc = activity.wc
        app = activity.app
    }
}